x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/components/actions_menu.tsx

```diff
@@ -205,6 +226,17 @@ export const AgentPolicyActionMenu = memo<{
                   <AgentEnrollmentFlyout agentPolicy={agentPolicy} onClose={onClose} />
                 </EuiPortal>
               )}
+              {isUpgradeAgentsModalOpen && (
+                <EuiPortal>
+                  <AgentUpgradeAgentModal
+                    agents={`policy_id: ${agentPolicy.id}`}
```

Comment:
Just a consideration, not strictly related to this change: the name of this prop is really confusing, it should be something like `agentIdsOrQuery`. The current name suggests that only `agent ids` can be passed. 😬

Reply:
Agreed, such a rename would be a good improvement 👍

Reply:
After some trying, I'd be inclined towards leaving that for another change if that's OK, the reason being this convention is used pretty much across this whole UI and it would involve a lot of files changed with respect to this otherwise very narrow change. Definitely making a note of it though.

Reply:
sure no worries, it's better to file a separate tech debt ticket for it